app/src/main/java/com/babylon/wallet/android/presentation/account/createaccount/CreateAccountViewModel.kt

```diff
@@ -141,15 +81,47 @@ class CreateAccountViewModel @Inject constructor(
                 discardTemporaryRestoredFileForBackupUseCase(BackupType.Cloud)
             }
 
-            if (state.value.useLedgerSelected) {
-                sendEvent(CreateAccountEvent.AddLedgerDevice(args.networkId))
+            val onNetworkId = if (args.networkId != -1) {
+                NetworkId.from(args.networkId)
             } else {
-                handleAccountCreate { name, networkId ->
-                    createAccountWithBabylonDeviceFactorSourceUseCase(
-                        displayName = name,
-                        networkID = networkId
-                    )
+                val profile = getProfileUseCase.invoke().first()
+                profile.currentNetwork?.knownNetworkId ?: Radix.Gateway.default.network.networkId()
+            }
+
+            // at the moment you can create a account either with device factor source or ledger factor source
+            var selectedFactorSource: FactorSource.CreatingEntity? = null
+
+            if (isWithLedger) { // get the selected ledger device
+                sendEvent(CreateAccountEvent.AddLedgerDevice)
+                selectedFactorSource = appEventBus.events
+                    .filterIsInstance<AppEvent.AccessFactorSource.SelectedLedgerDevice>()
+                    .first().ledgerFactorSource
+            }
+
+            // if main babylon factor source is not present, it will be created during the public key derivation
+            val publicKeyAndDerivationPath = accessFactorSourceProxy.getPublicKeyAndDerivationPathForFactorSource(
+                accessFactorSourceInput = AccessFactorSourceInput.ToCreateAccount(
+                    forNetworkId = onNetworkId,
+                    factorSource = if (isWithLedger && selectedFactorSource != null) {
+                        selectedFactorSource
+                    } else {
+                        null
+                    }
+                )
+            )
```

Comment:
actually here is the part that I don't like and we discussed about it in the chapter meeting.
What it should be ideal here is to pass an already created babylon device factor source and not to pass the null.